mesheryctl/internal/cli/root/system/check_test.go

```diff
@@ -0,0 +1,68 @@
+package system
+
+import (
+	"bytes"
+	"flag"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/sirupsen/logrus"
+)
+
+var update = flag.Bool("update", false, "update golden files")
+
+func TestDefaultPreflightCmd(t *testing.T) {
+	SetupContextEnv(t)
+	tests := []utils.CmdTestInput{
+		{
+			Name:             "Run preflight check",
+			Args:             []string{"check", "--pre"},
+			ExpectedResponse: "check.output.golden",
+		},
+		{
+			Name:             "Run preflight check",
+			Args:             []string{"check", "--preflight"},
+			ExpectedResponse: "check.output.golden",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.Name, func(t *testing.T) {
+			// setting up logrus to grab logs
+			var buf bytes.Buffer
+			logrus.SetOutput(&buf)
+
+			SystemCmd.SetArgs(tt.Args)
+			err = SystemCmd.Execute()
+			if err != nil {
+				t.Error(err)
+			}
+
+			output := buf.String()
+			actualResponse := utils.TrimLogOutputsTesting(output)
```

Comment:
> @leecalcote
> Unit tests check failures - -

> piyushsingariya
> Seems like we're not able to grab the logs here! What could be the possible reasons?? // @navendu-pottekkat

This error that you all are facing seems that the `utils.TrimLogOutputsTesting()` function is doing something wrong. If you assign the `output` variable as it is to the `actualResponse`, you'll see the logrus text:

```suggestion
			actualResponse := output
```

See:
![image](https://user-images.githubusercontent.com/2797052/122685155-53d78200-d1cf-11eb-8314-6a3f2dd2fcc2.png)